kernelci/cli/base.py

```diff
@@ -620,14 +620,20 @@ def get_from_section(self, section, option, as_list=False):
         *as_list* is like for .get() for options with multiple values
         """
         if self._deprecated_settings:
+            if isinstance(section, tuple):
+                section = ':'.join([section[0], section[1]])
             if not self._settings.has_option(section, option):
                 return None
             value = self._settings.get(section, option).split()
             if not as_list and len(value) == 1:
                 value = value[0]
         else:
             value = None
-            section_data = self._settings.get(section)
+            if isinstance(section, tuple):
+                section_data = self._settings.get(
+                    section[0], {}).get(section[1])
```

Comment:
I think we should be able to just stop using `get_from_section()` when the old settings format is dropped.  The code could just look up the sub-section itself, for example with `--api-config=something` it could get the `api` section from the settings and then look for the `something`  key in the application as the information is already there.  Until now, this used to be "serialised" as `section:subsection` but now we don't need that any more.

Until the old settings format is deprecated, I think it's going to be difficult to manage that.  But maybe the new code based on the `kci` command line tool and the new pipeline services could already require the settings file to be TOML and make use of the dotted-key sections natively.

Reply:
Yes, we would be able to drop the `get_from_section` method and move its whole logic to `Options.get` once the old settings are dropped.